### PR TITLE
🛠️ `Neighborhood`: Support `bin/erd` and `yarn test` in devcontainer

### PR DESCRIPTION
**Pull request by daltonrpruitt**
Graphviz for ERD generation. 
Firefox for browser/front-end testing.
Xvfb for virtual display (headless mode on firefox). 
Set DISPLAY environment variable for `firefox` to know what to use. 

Co-authored-by: Zee Spencer <zspencer@users.noreply.github.com>
Co-authored-by: Kelly Hong <KellyAH@users.noreply.github.com>
Co-authored-by: Ana <anaulin@users.noreply.github.com>

### CODE REVIEW

**Comment by daltonrpruitt** [comment]
TODO: Need to either
1. Update ruby version of project to 3-ish
2. Install older ruby version in devcontainer
Currently, the version is being update, so, this can probably be rebased off of the updated `main` and it should just work.

**Comment by daltonrpruitt** [comment]
The ruby version we use was updated, so the above TODO was taken care of. 
The `universal` devcontainer image was too large for our tastes, so we decided to go back to ruby container directly. 
(We may want to look into the features stuff for devcontainers. Not currently sure what that does.)

The current version (10 commits, just removed the `gem install` from the Dockerfile) may work, but I'm having trouble running `yarn run test` and haven't dug through the docs to verify I'm set up for that yet. Will look at later.

**Comment by daltonrpruitt** [comment]
Currently running into an error when running `yarn run test` (I assumed was how to run the cucumber tests).
Essentially, starting up this version of the devcontainer in a codespace, verifying convene is running by forwarding port 3000 (why it's not being done automatically anymore is unknown), and then running `yarn run test` results in the following:

```bash
@daltonrpruitt ➜ /workspaces/convene (add_graphviz_to_devcontainer) $ yarn run test
yarn run v1.22.19
$ cucumber-js --format @cucumber/pretty-formatter --tags 'not @unscheduled and not @wip and not @unimplemented-steps and not @unstarted' --import features/
VError: a BeforeAll hook errored, process exiting: file:/workspaces/convene/features/support/env.js:29: Process unexpectedly closed with status 1
    at Runtime.runTestRunHooks (/workspaces/convene/node_modules/@cucumber/cucumber/lib/runtime/run_test_run_hooks.js:23:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Runtime.start (/workspaces/convene/node_modules/@cucumber/cucumber/lib/runtime/index.js:53:9)
    at async runCucumber (/workspaces/convene/node_modules/@cucumber/cucumber/lib/api/run_cucumber.js:106:21)
    at async Cli.run (/workspaces/convene/node_modules/@cucumber/cucumber/lib/cli/index.js:50:29)
    at async Object.run [as default] (/workspaces/convene/node_modules/@cucumber/cucumber/lib/cli/run.js:34:18)
caused by: WebDriverError: Process unexpectedly closed with status 1
    at Object.throwDecodedError (/workspaces/convene/node_modules/selenium-webdriver/lib/error.js:524:15)
    at parseHttpResponse (/workspaces/convene/node_modules/selenium-webdriver/lib/http.js:587:13)
    at Executor.execute (/workspaces/convene/node_modules/selenium-webdriver/lib/http.js:515:28)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I have tried to look into things, but I feel that someone may have seen something like this before and I don't my debugging of this to be the hold up when I have confirmed that the two packages were installed (can run `bin/erd` and can run `firefox` (which [fails successfully](https://www.youtube.com/watch?v=Si8C0djBi-0) ).

**Comment by zspencer** [comment]
Turns out, this error is caused by firefox not having a X window system to render to. It is possible to solve by making sure `xvfb` is running (i.e. `xvfb :1` and that a `DISPLAY` environment variable points to the virtual frame buffer.

https://griggheo.medium.com/running-selenium-webdriver-tests-using-firefox-headless-mode-on-ubuntu-d32500bb6af2

**Comment by daltonrpruitt** [comment]
Finally got it to run `yarn test` in a fresh devcontainer. I hope these commits are squahed in the merge...

File: .devcontainer/startup.bash [hunk 1]
```diff
@@ -7,14 +7,16 @@ docker compose up &> .devcontainer/output/docker_compose_up.out &
...
 
-if [ ! -f .env ]; then 
-    cp .env.example .env 
-    sed -i "/^# PG/s/^# //g" .env 
-fi 
+Xvfb $DISPLAY -ac &> .devcontainer/output/Xvfb.out &
+
+if [ ! -f .env ]; then
+    cp .env.example .env
+    sed -i "/^# PG/s/^# //g" .env
```
Comment (zspencer) [inline .devcontainer/startup.bash:14]: Oh damn `sed`

**Review by zspencer — APPROVED** [review]
This rocks!! Thank you!!!

File: .devcontainer/startup.bash (hunk 1 above)
Comment (daltonrpruitt) [reply .devcontainer/startup.bash:14]: Seemed like the easiest way to setup the env file for docker stuff. I'm just glad it works.

**Pull request merged**